src/onINP.ts

```diff
@@ -57,20 +63,88 @@ const getInteractionCountForNavigation = () => {
 const MAX_INTERACTIONS_TO_CONSIDER = 10;
 
 // A list of longest interactions on the page (by latency) sorted so the
-// longest one is first. The list is as most MAX_INTERACTIONS_TO_CONSIDER long.
+// longest one is first. The list is at most MAX_INTERACTIONS_TO_CONSIDER long.
 let longestInteractionList: Interaction[] = [];
 
 // A mapping of longest interactions by their interaction ID.
 // This is used for faster lookup.
-const longestInteractionMap: {[interactionId: string]: Interaction} = {};
+const longestInteractionMap: {[interactionId: string]: Interaction} =
+  Object.create(null);
+
+// A mapping between a particular frame's render time and all of the
+// event timing entries that occurred within that frame. Since event timing
+// entries round their duration to the nearest 8sm, this ends up being
+// a best-effort guess, but for debugging and attribution practices it's
+// useful to see all events the occurred within the same frame as a particular
+// interaction, so it's worth doing the best-effort guess.
+// const pendingEntriesGroupMap: Map<number, PendingEntriesGroup> = new Map();
+const pendingEntriesGroupMap: {[renderTime: string]: PendingEntriesGroup} =
+  Object.create(null);
+
+/**
+ * Gets the "frame-normalized" render time for the given entry.
+ * In other words, all entries passed to this function that were processed
+ * within the same frame should return the same render time. This allows you
+ * to group by this render time value.
+ * This function works by referencing `pendingEntriesGroupMap` and using
+ * an existing render time if one is found (otherwise creating a new one).
+ */
+const groupEntryByRenderTime = (entry: PerformanceEventTiming) => {
```

Comment:
Nit: you might want to make a distinction between event entry renderTime, and animation frame estimated renderTime (or eventGroup estimated renderTime).

One entry at a time it's not so bad, but when you overwrite the current entry renderTime with the frame estimated renderTime, you start to get values that might be surprisingly off by a bit.

Also, estimated frame render time could be slightly better than just the "first entry renderTime"

Reply:
The `renderTime` values here are just used for grouping, they aren't exposed anywhere in the `attribution` object.